runtime/tests/interpreter/member_test.go

```diff
@@ -0,0 +1,523 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2022 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package interpreter_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/tests/checker"
+)
+
+func TestInterpretMemberAccessType(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("direct", func(t *testing.T) {
+
+		t.Run("non-optional", func(t *testing.T) {
+
+			t.Run("valid", func(t *testing.T) {
+
+				t.Parallel()
+
+				inter := parseCheckAndInterpret(t, `
+                    struct S {
+                        var foo: Int
+
+                        init() {
+                            self.foo = 1
+                        }
+                    }
+
+                    fun get(s: S) {
+                        s.foo
+                    }
+
+                    fun set(s: S) {
+                        s.foo = 2
+                    }
+                `)
+
+				value, err := inter.Invoke("S")
```

Comment:
What does invoking "S" do? Does it directly calls the Constructor of "S"?

Reply:
Yes, declaring a composite declares a value for the constructor, a function with the name of the type. I'll add a comment